target-chains/cosmwasm/contracts/pyth/src/contract.rs

```diff
@@ -423,42 +439,19 @@ fn update_price_feed_if_new(
 }
 
 #[cfg_attr(not(feature = "library"), entry_point)]
-pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
+pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
     match msg {
-        QueryMsg::PriceFeed { id } => to_binary(&query_price_feed(&deps, env, id.as_ref())?),
+        QueryMsg::PriceFeed { id } => to_binary(&query_price_feed(&deps, id.as_ref())?),
         QueryMsg::GetUpdateFee { vaas } => to_binary(&get_update_fee(&deps, &vaas)?),
         QueryMsg::GetValidTimePeriod => to_binary(&get_valid_time_period(&deps)?),
     }
 }
 
-pub fn query_price_feed(deps: &Deps, env: Env, address: &[u8]) -> StdResult<PriceFeedResponse> {
-    let config = config_read(deps.storage).load()?;
+pub fn query_price_feed(deps: &Deps, address: &[u8]) -> StdResult<PriceFeedResponse> {
     match price_info_read(deps.storage).load(address) {
-        Ok(mut price_info) => {
-            let env_time_sec = env.block.time.seconds();
-            let price_pub_time_sec = price_info.price_feed.publish_time as u64;
-
-            // Cases that it will cover:
-            // - This will ensure to set status unknown if the price has become very old and hasn't
-            //   updated yet.
-            // - If a price has arrived very late to this chain it will set the status to unknown.
-            // - If a price is coming from future it's tolerated up to VALID_TIME_PERIOD seconds
-            //   (using abs diff) but more than that is set to unknown, the reason could be the
-            //   clock time drift between the source and target chains.
-            let time_abs_diff = if env_time_sec > price_pub_time_sec {
-                env_time_sec - price_pub_time_sec
-            } else {
-                price_pub_time_sec - env_time_sec
-            };
-
-            if time_abs_diff > config.valid_time_period.as_secs() {
```

Comment:
so is the intended use here that if people want to use the default valid time period, they can query the contract to retrieve the configured value, then pass that to `get_price_no_older_than`?

Reply:
Yes, it is upto them how they want to validate the price. They can fetch the default one. 
@ali-bahjati, can you please confirm this?

Reply:
That's right. To my experience people usually don't use default time and rather use their own desired time. However, the problem about this was that the pyth-sdk has no clue about the `valid_time_period`. We are planning to remove the `pyth-sdk` entirely in the future and have per chain PriceFeed definition.

Also in other platforms we have more query methods, like `get_price` `get_price_no_older_than` `get_price_unsafe`. We can implement them in the future here if we realize the query of PriceFeed is expensive.